test/passing/dune

```diff
@@ -16,19 +15,54 @@ let read_lines fn =
 let register_file tests fname =
   match String.split_on_char '.' fname with
   | test_name :: (("ml" | "mli") as ext) :: rest -> (
-      let test_name = test_name ^ "." ^ ext in
+      let src_test_name = test_name ^ "." ^ ext in
       let setup =
-        match Hashtbl.find tests test_name with
+        match Hashtbl.find tests src_test_name with
         | setup -> setup
-        | exception Not_found ->
-            let s = {has_ref= false; has_opts= false; extra_deps= []} in
-            Hashtbl.add tests test_name s ;
-            s
+        | exception Not_found -> (
+          (* foo_file-some_variant.ml should derive from foo_file.ml *)
+          match String.index_opt test_name '-' with
+          | None ->
+              let s = {outputs= []; extra_deps= []} in
+              Hashtbl.add tests src_test_name s ;
+              s
+          | Some i -> (
+              let src_test_name = String.sub test_name 0 i ^ "." ^ ext in
+              match Hashtbl.find tests src_test_name with
+              | setup -> setup
+              | exception Not_found ->
+                  let s = {outputs= []; extra_deps= []} in
+                  Hashtbl.add tests src_test_name s ;
+                  s ) )
       in
       match rest with
       | [] -> ()
-      | ["opts"] -> setup.has_opts <- true
-      | ["ref"] -> setup.has_ref <- true
+      | ["opts"] ->
+          let ref_name = test_name ^ "." ^ ext ^ ".ref" in
+          let ref_exists {ref; _} =
+            Option.equal String.equal ref (Some ref_name)
+          in
+          let outputs =
+            if List.exists ref_exists setup.outputs then
+              let opts = Some fname in
+              let update o = if ref_exists o then {o with opts} else o in
+              List.map update setup.outputs
+            else {ref= None; opts= Some fname} :: setup.outputs
+          in
```

Comment:
This code looks too complicated. Is it necessary to pair `ref` and `opts`, even when one can be `None` ? Can't each variations of a tests be in the `tests` table and having a single `out` ?

Reply:
I reduced the diff.

Reply:
I now have a lot of errors like
`Error: No rule found for test/passing/break_before_in.ml.ref (context
coverage)`
A lot of them is because the rule is using `.ref` or `.opts` files that don't exists

Reply:
It should be fixed now, I forgot to clean the env before testing this locally.